File: Python/78.子集.py
from typing import List

class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        l = len(nums)
        maxn = pow(2, l)
        ans: List[List[int]] = [[] for i in range(maxn)]
        for i in range(maxn):
            for j in range(l):
                if ((i >> j) & 1) == 1:
                    ans[i].append(nums[j])
        return ans

File: 78.子集.cc
#include <vector>
#include <cmath>
using std::vector;
using std::pow;

class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int len = nums.size();
        int maxn = pow(2, len);
        vector<vector<int>> ans(maxn);
        for (int i = 0; i < maxn; ++i)
            for (int j = 0; j < len; ++j)
                if ((i >> j) & 1)
                    ans[i].push_back(nums[j]);
        return ans;
    }
};
